ament_cmake_core/cmake/environment_hooks/ament_generate_package_environment.cmake

```diff
@@ -49,7 +49,10 @@ function(ament_generate_package_environment)
 
       # collect package hooks to be sourced for this extension
       set(ENVIRONMENT_HOOKS "")
-      if(NOT "${_AMENT_CMAKE_ENVIRONMENT_HOOKS_${extension}}" STREQUAL "")
+      if(
+        DEFINED _AMENT_CMAKE_ENVIRONMENT_HOOKS_${extension} AND
+        NOT "${_AMENT_CMAKE_ENVIRONMENT_HOOKS_${extension}}" STREQUAL ""
```

Comment:
This part of the condition is redundant. Removed in 6d171d79584319c216e6880c46372224cd3ae6eb.